### PR TITLE
SIP 4.19.7+ allows to document template based classes

### PR DESCRIPTION
**Pull request by 3nids**
it'll be a bit tricky to bring this in the documentation

* dealing with this in sipify is wrong since the source must be independant of the compiling tools (i.e. same source for all)
* sip does not have any preprocessing
* %If directive leads to a syntax error (code gets evaluated even if the condition is not entered)

The only solution I see for the moment is to have an extra option in sipify to output the Docstring where wanted and to rewrite all the SIP files when building the API doc. We will keep the sip files without Docstrings for template based classes in the main QGIS sources. 

so to create these, one needs to run
`./scripts/sipify_all -template-doc`

### CODE REVIEW

File: python/core/qgsrange.sip [hunk 1]
```diff
@@ -246,9 +246,9 @@ Returns true if this range overlaps another range.
 };
 
 
-typedef QgsTemporalRange< QDate > QgsDateRange;
+typedef QgsTemporalRange< QDate > QgsDateRange ;
```
Comment (nirvn) [inline python/core/qgsrange.sip]: Unwanted extra space?

**Comment by 3nids** [comment]
I decided to go for the input configuration files for all auto-generated sip files.
This way, the code is properly hosted on github with all features.

File: python/core/qgsrange.sip (hunk 1 above)
Comment (3nids) [reply python/core/qgsrange.sip]: ok, fixed

**Comment by 3nids** [comment]
anybody against the idea of having input files for all sip files in source?

**Comment by nyalldawson** [comment]
@3nids what's the new approach for updating sip bindings? I noticed that the prepare_commit script no longer works since this change

**Comment by 3nids** [comment]
should be fixed now, you'll have to rebase.
sorry for that!

**Pull request merged**